rostest/scripts/main.py

```diff
@@ -28,54 +32,36 @@ def change_frequency(new_frequency):
     period = 1.0 / run_frequency
     return new_frequencyResponse("We received you frequency!")
 
-
 def run(showgui: bool):
-    pub = rospy.Publisher("observations", observation, queue_size=50)
     message_service = rospy.Service('inputs', text_message, print_input)
     frequency_service = rospy.Service('frequency', new_frequency,
                                       change_frequency)
-    cam = cv.VideoCapture(source)
-    detector = ObjectDetector("ssd_mobilenet_v1_1_metadata_1.tflite",
-                              "mscoco_complete_labels")
+    # Image feed topic
+    rospy.Subscriber("camera_feed", image, detect)
+
+def detect(img):
+    start_time = time.time()
+    # Converting the image back from ros Image message to a numpy ndarray
+    img = converter.msg_to_cv2(img)
+    # Resizing could be better to do before sending the message, to save a little bandwidth
+    inp = cv.resize(img, (300, 300))
```

Comment:
Dependent on the model used, some common models use 280x280 instead etc.
Some alternative way of defining dimensions for resize should be used.

Reply:
Does the model contain information of required dimensions?